phpjs.c

```diff
@@ -1,317 +1,369 @@
+{
+    /*
+    DONE : IS_NULL     0
+    DONE : IS_LONG     1
+    DONE : IS_DOUBLE   2
+    DONE : IS_BOOL     3
+    DONE : IS_ARRAY    4
+    INPROGRESS : IS_OBJECT   5
+    DONE : IS_STRING   6
+    DONE : IS_RESOURCE 7
+    TODO : IS_CONSTANT 8
+    TODO : IS_CONSTANT_ARRAY   9
+    TODO : IS_CALLABLE 10
+    */
+    switch (Z_TYPE_P(value)) {
+    case IS_ARRAY: {
+        zval ** data;
+        HashTable *myht = Z_ARRVAL_P(value);
+        char *str_index;
+        uint str_length;
+        ulong num_index;
+        duk_idx_t arr_idx = duk_push_php_array_or_object(ctx, myht);
+
+        for (zend_hash_internal_pointer_reset(myht);
+               zend_hash_get_current_data(myht, (void **) &data) == SUCCESS;
+               zend_hash_move_forward(myht)
+           ) {
+            zval_to_duk(ctx, NULL, *data);
+            switch (zend_hash_get_current_key_ex(myht, &str_index, &str_length, &num_index, 0, NULL)) {
+            case HASH_KEY_IS_LONG:
+                duk_put_prop_index(ctx, arr_idx, num_index);
+                break;
+            case HASH_KEY_IS_STRING:
+                duk_put_prop_string(ctx, arr_idx, str_index);
+                break;
+            }
+
+        }
+        break;
+    }
+    case IS_STRING:
+        duk_push_string(ctx, Z_STRVAL_P(value));
+        break;
+    case IS_LONG:
+        duk_push_int(ctx, Z_LVAL_P(value));
+        break;
+    case IS_DOUBLE:
+        duk_push_number(ctx, Z_DVAL_P(value));
+        break;
+    case IS_BOOL:
+        if (Z_BVAL_P(value)) {
+            duk_push_true(ctx);
+        } else {
+            duk_push_false(ctx);
+        }
+        break;
+    case IS_NULL:
+        duk_push_null(ctx);
+        break;
+    case IS_RESOURCE:
+        duk_push_pointer(ctx, Z_RESVAL_P(value));
+        break;
+    case IS_OBJECT: {
+            zval *val;
+            MAKE_STD_ZVAL(val);
+            TSRMLS_FETCH();
+            ZVAL_ZVAL(val,value,1,0);
+            if(zend_is_callable(val, 0, NULL TSRMLS_CC)){
+                duk_push_c_function(ctx, php_object_handler, DUK_VARARGS);
+                duk_push_string(ctx, "__invoke");
+                duk_put_prop_string(ctx, -2, "__function");
+                duk_push_pointer(ctx, val);
+                duk_put_prop_string(ctx, -2, "__zval__");
+                // * /                
+            }else{
+                // create a proxy for handle value and method
+                duk_int_t rc,args = 1;
+                char * className ;
+                zend_uint classNameLength;  
+                duk_push_string(ctx, "(function(r,c) { return Duktape.initPHPObj(r,c); })");
+                duk_peval(ctx); 
+                duk_push_int(ctx, LAMBDA_CALLBACK_INDEX++);
+                duk_push_pointer(ctx, val);
+                // TODO : Get Class Name
+                int ret =FAILURE;
+                ret = zend_get_object_classname(value,&className,&classNameLength);

```

Comment:
`/home/travis/build/crodas/PHPJS/phpjs.c:141:17: error: too few arguments to function ‘zend_get_object_classname’`

I believe it should be:

```c
ret = zend_get_object_classname(value,&className,&classNameLength TSRMLS_CC);
```